File: Daily-Streak/Day-01_Plus-One/Solution.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        n = len(digits)

        # Traverse from the last digit
        for i in range(n - 1, -1, -1):
            digits[i] += 1

            # If no carry, return immediately
            if digits[i] < 10:
                return digits

            # Carry case
            digits[i] = 0

        # If all digits were 9, add 1 at the beginning
        return [1] + digits

File: Daily-Streak/Day-01_Plus-One/Solution.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;

        // Traverse from the last digit
        for (int i = n - 1; i >= 0; i--) {
            // Add one to the current digit
            digits[i]++;

            // If no carry is generated, return result
            if (digits[i] < 10) {
                return digits;
            }

            // Carry case
            digits[i] = 0;
        }

        // If all digits were 9, create new array
        int[] result = new int[n + 1];
        result[0] = 1;
        return result;
    }
}
